src/var.cc

```diff
@@ -151,7 +150,9 @@ RecursiveVar::RecursiveVar(Value* v,
 
 void RecursiveVar::Eval(Evaluator* ev, string* s) const {
   ev->CheckStack();
+  ev->AddVarToEvalStack(this);
```

Comment:
Why not raise the "variable is self-referential" error here? Then one needs to look at fewer places to understand how self-reference detection works. E.g. you could make `AddVarToEvalStack()` return a Boolean or make it signal an error itself.

Reply:
That would be preferable, but unfortunately it's not until the value is marked as used that we get the symbol that we need to use in the error -- I didn't see a straightforward way to pipe through that information without a lot of changes.